### PR TITLE
Skip mutations

### LINKED ISSUE #26: Skip the cache if the request is a mutation
When a mutation is done, the user can add a query in the request.

It will be useful to verify the type of request and skip the cache when is a `mutation`.

In this way the user will get always data up to date after a modify action.

### PR DESCRIPTION
**Pull request by sameer-coder**
This PR makes the cache skip mutations

Closes #26

### CODE REVIEW

**Comment by mcollina** [comment]
cc @simone-sanfratello wdyt?

File: index.js [hunk 1]
```diff
@@ -124,7 +124,7 @@ function makeCachedResolver (prefix, fieldName, cache, originalFieldResolver, po
   })
   return async function (self, arg, ctx, info) {
     if ((skip && await skip(self, arg, ctx, info)) ||
-      (policy && policy.skip && await policy.skip(self, arg, ctx, info))) {
+      (policy && policy.skip && await policy.skip(self, arg, ctx, info)) || info?.operation?.operation === 'mutation') {
```
Comment (mcollina) [inline index.js:127]: I have a feeling that ? is not available in Node v12.

Anyhow could you move the check before the policy check?

Comment (sameer-coder) [reply index.js]: You are right. I checked and `?` doesn't work on node 12. I have removed it now and also moved the check before policy check

File: index.js [hunk 2]
```diff
@@ -124,6 +124,7 @@ function makeCachedResolver (prefix, fieldName, cache, originalFieldResolver, po
   })
   return async function (self, arg, ctx, info) {
     if ((skip && await skip(self, arg, ctx, info)) ||
+      (info.operation && info.operation.operation === 'mutation') ||
```
Comment (simone-sanfratello) [inline index.js:127]: in this way, the "event" `onSkip` is called, but it shouldn't
imo the mutation skip should just call the original resolver

**Review by simone-sanfratello — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (sameer-coder) [reply index.js:127]: yeah I was thinking about that too but wasn't sure if `onSkip` should be called or not

Comment (simone-sanfratello) [reply index.js]: I'd go for something like this

```js
if(info.operation && info.operation.operation === 'mutation') {
  return originalFieldResolver(self, arg, ctx, info)
}

if ((skip && await skip(self, arg, ctx, info)) ||
  (policy && policy.skip && await policy.skip(self, arg, ctx, info))) {
  onSkip()
// ...
```

**Review by mcollina — APPROVED** [review]
lgtm

**Review by simone-sanfratello — APPROVED** [review]

**Pull request merged**